streaming/base/converters/dataframe_to_mds.py

```diff
@@ -287,6 +291,17 @@ def write_mds(iterator: Iterable):
 
     # Fix output format as mds_path: Tuple(local, remote)
     if cu.remote is None:
+        # If dataframe_to_mds is being called, this is in a distributed Spark env.
```

Comment:
@XiaohanZhangCMU is this true? I can call dataframe_to_mds from outside a distributed spark env right?

I don't mind the warning, I think it's useful. Just want to check if this comment was right.

Reply:
The method is definitely specific to Spark. But Spark can be run on one machine, yeah. If one passed a local path as 'out' in this case, then that would be fine. Setting ("/tmp/...", "/my/local/path/...") instead would also work; it would just be a bit redundant - it would save to a local path, then copy to another local path.

It would not make sense to path a local path if running on a cluster of machines, as the local path is specific to each individual machine.

Even if running on a single-machine, the storage could be remote. That case is also fine, as CloudUploader will understand the single remote path argument as a remote.

The case here is, really, where you have remote storage that looks like a local path. That arises when you have FUSE mounted storage, and that is definitely how Databricks _can_ work. Then we have a little problem of a lot of little writes to a 'local' path that is actually S3 or similar. It still 'works', just, can be inefficient or hit rate limits.


I think the intended message is: if you pass what looks like a local path here, it's either:
- OK, because this is a single machine though you're using Spark anyway
- Wrong, because you're not referencing a FUSE mount and the files aren't going where you think they are
- OK but inefficient because it's FUSE mounted storage

I can try to think of a rewording here. 
If you passed a 'local' path, warn that this is only correct if running Spark on a single machine, or if the storage is a FUSE mount. In the latter case you want to specify a local temp path, ideally.

But the running Spark on one machine case seemed rare, and, setting a local temp path is OK in that case anyway. So, maybe the simplest guidance is:

Please set a 'local' temp path too, even if your intended 'remote' destination is also a local path

Keep it simple?

Reply:
Sure yeah. sounds good to me. just was in doubt about the wording.